tests/hazmat/primitives/test_dh.py

```diff
@@ -102,3 +104,217 @@ def test_dh_public_numbers_equality():
     assert public != dh.DHPublicNumbers(0, params)
     assert public != dh.DHPublicNumbers(1, dh.DHParameterNumbers(65537, 0))
     assert public != object()
+
+
+@pytest.mark.requires_backend_interface(interface=DHBackend)
+class TestDH(object):
+    def test_small_key_generate_dh(self, backend):
+        with pytest.raises(ValueError):
+            dh.generate_parameters(2, 511, backend)
+
+    def test_dh_parameters_supported(self, backend):
+        assert backend.dh_parameters_supported(23, 5)
+        assert not backend.dh_parameters_supported(23, 18)
+
+    def test_convert_to_serialized(self, backend):
+        parameters = backend.generate_dh_private_key_and_parameters(2, 512)
+
+        private = parameters.private_numbers()
+
+        p = private._public_numbers._parameter_numbers.p
```

Comment:
`public_numbers` and `parameter_numbers` should be accessible without an underscore (on this and subsequent lines)